src/gax-internal/src/observability.rs

```diff
@@ -432,4 +413,154 @@ mod tests {
+            let span = span_info.create_span();
+            span.id().unwrap()
+        });
+
+        let spans = layer.spans.lock().unwrap();
+        let attributes = spans.get(&id).expect("span not found");
+
+        // Check all attributes set in create_span
+        assert_eq!(attributes.len(), 15, "Unexpected number of attributes");
+        assert_eq!(attributes.get(KEY_OTEL_NAME).unwrap(), "GET /test");
+        assert_eq!(attributes.get(KEY_OTEL_KIND).unwrap(), "Client");
+        assert_eq!(attributes.get(otel_trace::RPC_SYSTEM).unwrap(), "http");
+        assert_eq!(
+            attributes.get(otel_trace::HTTP_REQUEST_METHOD).unwrap(),
+            "GET"
+        );
+        assert_eq!(
+            attributes.get(otel_trace::SERVER_ADDRESS).unwrap(),
+            "example.com"
+        );
+        assert_eq!(attributes.get(otel_trace::SERVER_PORT).unwrap(), "443");
+        assert_eq!(
+            attributes.get(otel_trace::URL_FULL).unwrap(),
+            "https://example.com/test"
+        );
+        assert_eq!(attributes.get(otel_trace::URL_SCHEME).unwrap(), "https");
+        assert_eq!(attributes.get(otel_attr::URL_TEMPLATE).unwrap(), "/test");
+        assert_eq!(
+            attributes.get(otel_attr::URL_DOMAIN).unwrap(),
+            "example.com"
+        );
+        assert_eq!(
+            attributes.get(KEY_GCP_CLIENT_SERVICE).unwrap(),
+            "test.service"
+        );
+        assert_eq!(attributes.get(KEY_GCP_CLIENT_VERSION).unwrap(), "1.2.3");
+        assert_eq!(
+            attributes.get(KEY_GCP_CLIENT_REPO).unwrap(),
+            "googleapis/google-cloud-rust"
+        );
+        assert_eq!(
+            attributes.get(KEY_GCP_CLIENT_ARTIFACT).unwrap(),
+            "google-cloud-test"
+        );
+        assert_eq!(
+            attributes
+                .get(otel_trace::HTTP_REQUEST_RESEND_COUNT)
+                .unwrap(),
+            "1"
+        );
+    }
+
+    #[tokio::test]
+    async fn test_create_span_attributes_optional() {
+        let layer = TestLayer::default();
+        use tracing_subscriber::prelude::*;
+        let subscriber = tracing_subscriber::registry().with(layer.clone());
+
+        let id = tracing::subscriber::with_default(subscriber, || {
+            let request =
+                reqwest::Request::new(Method::POST, "http://localhost:8080/".parse().unwrap());
+            let options = RequestOptions::default(); // No path template
+            // No InstrumentationClientInfo
+            let span_info = HttpSpanInfo::from_request(&request, &options, None, 0);
+            let span = span_info.create_span();
+            span.id().unwrap()
+        });
+
+        let spans = layer.spans.lock().unwrap();
+        let attributes = spans.get(&id).expect("span not found");
+
+        // Check only attributes that should be None or not present
+        assert_eq!(attributes.len(), 9, "Unexpected number of attributes");
```

Comment:
That literal bother me, can we turn this into something more explicit? Maybe check the expected keys?